pkg/runtime/adoption_reconciler.go

```diff
@@ -85,6 +85,18 @@ func (r *adoptionReconciler) reconcile(req ctrlrt.Request) error {
 	}
 
 	gk := r.getTargetResourceGroupKind(res)
+
+	// Check if the target API group matches with the controller
+	var controllerRMF acktypes.AWSResourceManagerFactory
+	for _, v := range r.sc.GetResourceManagerFactories() {
+		controllerRMF = v
+		break
+	}
+	if gk.Group != controllerRMF.ResourceDescriptor().GroupKind().Group {
+		LogAdoptedResource(r.log, res).V(0).Info("target resource API group is not of this service. no-op")
```

Comment:
I'd make this into a `.Debug`. I'm really trying to avoid spamming logs with information, especially if this is an intentional no-op.